core/primitives/src/rpc.rs

```diff
@@ -8,4 +9,13 @@ pub struct RpcQueryRequest {
     pub block_id: MaybeBlockId,
     #[serde(flatten)]
     pub request: QueryRequest,
+    pub finality: Finality,
```

Comment:
We need a default here as otherwise, it will be a breaking change. I suggest you implement the `Default` trait by hands and I will refactor it after #2109 (I add `smart-default` there), or you introduce `smart-default` and I will rebase my PR.

Our test suite is useless for this type of API changes, I need to do a better test suite.